### PR TITLE
fix #138 countries graphql errors on explode

### LINKED ISSUE #138: Countries GraphQL query throws error over explode function in getStateRequiredCountries(): array
Upon installing the latest version of the checkout into a magento 2.3.5-p2 project we encountered a bug in the `countries` GraphQL Query.

query send:
```graphql
{
  countries {
    id
  }
}
```

result:
```json
{
  "errors": [
    {
      "debugMessage": "explode() expects parameter 2 to be string, null given",
      "message": "Internal server error",
      "extensions": {
        "category": "internal"
      },
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "countries"
      ]
    }
  ],
  "data": {
    "countries": null
  }
}
```

expected:
```json
{
  "data": {
    "countries": [
      {
        "id": "BE"
      },
      {
        "id": "NL"
      }
    ]
  }
}
```

The culprit appears to be within the `getStateRequiredCountries(): array` method, inside `Hyva/Checkout/Plugin/DirectoryModel/SetStateRequired.php`. Here the explode function fails and throws the error because in our project the configuration string `'general/region/state_required'` doesnt exist.

I will open up a PR adding a simple if check to test if the configuration doesnt exist and return an empty array instead of erroring out. However I also believe we should somewhere document the dependency on this configuration setting and how developers should configure it appropriately. This I shall leave up to you to decide.

### PR DESCRIPTION
**Pull request by peterschriever**
Hello, this PR fixes the throwing of an error when `general/region/state_required` isnt configured or is empty. In such cases it would now return an empty array for the countries of which state should be required.

We still need to document how and that developers should appropriately configure this option though!

as described in #138

### CODE REVIEW

File: src/Plugin/DirectoryModel/SetStateRequired.php [hunk 1]
```diff
@@ -113,11 +113,13 @@ private function getStateRequiredCountries(): array
...
         $store = $this->getStore();
         if (!$this->stateRequiredCountries) {
-            $this->stateRequiredCountries = explode(',', $this->scopeConfig->getValue(
+            $configStateRequired = $this->scopeConfig->getValue(
                 'general/region/state_required',
                 ScopeInterface::SCOPE_STORES,
                 $store->getCode()
-            ));
+            );
+            if (!$configStateRequired) return [];
```
Comment (schmengler) [inline src/Plugin/DirectoryModel/SetStateRequired.php:121]: That will work but let's always save the value in `$this->stateRequiredCountries`

A more simple fix would be:
```
            $this->stateRequiredCountries = explode(',', (string) $this->scopeConfig->getValue(
                'general/region/state_required',
                ScopeInterface::SCOPE_STORES,
                $store->getCode()
            ));
```

no additional variables, still a single return point

**Review by schmengler — CHANGES_REQUESTED** [review]

**Comment by peterschriever** [comment]
Do u mean putting an empty array inside `$this->stateRequiredCountries` ? And then having the explode in an else?

The reason explode throws an error here is because '$this->scopeConfig->getValue' returns a null value. Even with the `(string)` tag i believe. Another option would be a try catch, but i thought an early return to be the cleanest and most simple.

**Comment by progammer-rkt** [comment]
> Do u mean putting an empty array inside `$this->stateRequiredCountries` ? And then having the explode in an else?
> 
> The reason explode throws an error here is because '$this->scopeConfig->getValue' returns a null value. Even with the `(string)` tag i believe. Another option would be a try catch, but i thought an early return to be the cleanest and most simple.

@peterschriever  The error in the issue says: `explode() expects parameter 2 to be string, null given`. So turning `null` to an empty string would work. So `(string)` will work for sure. But still the suggested solution will provide `['']` as its ouput. ie, array is not really empty. So the correct code would be:

```
$this->stateRequiredCountries = array_filter(explode(',', (string) $this->scopeConfig->getValue(
                'general/region/state_required',
                ScopeInterface::SCOPE_STORES,
                $store->getCode()
            )));
```

**Comment by peterschriever** [comment]
@progammer-rkt @schmengler I've finally found time to update this PR, implemented the last suggestion from Rajeev.

**Review by progammer-rkt — APPROVED** [review]
Looks good now. Thank you for this PR!
